### PR TITLE
Integrate cosign - an implementation of the update framework in ballerina distribution releases

### LINKED ISSUE #4550: Integrate coisgn - implementation of the update framework in Ballerina Distribution Releases
**Description:**
Integrating a implementation of The Update Framework (TUF), this can be done via `Sigstore's Cosign` Tool. `Cosign` can be used to sign and verify the release artifacts, in this case would be to sign and release the Ballerina Releases. This would further provide user's the confidence that Ballerina is improving the security in its supply chain.

**Describe your problem(s)**

Ballerina may be vulnerable to supply chain attacks. If a release of Ballerina is compromised, and its passed on to the user, this will negatively affect his/her experience and thereby hinder the adoption of Ballerina.

**Describe your solution(s)**

Use the `Cosign` tool along with the other tools from ` Sigstore` to sign the artifacts being released, the user can thereafter verify it on his/her end, and we can include auto verification via `Update Tool` using `Sigstore's` Transparency Log `Rekor`.

### PR DESCRIPTION
**Pull request by Miranlfk**
## Purpose
$title

## Goals
Fixes #4550

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -69,7 +73,7 @@ jobs:
           devCentralToken: ${{ secrets.BALLERINA_DEV_CENTRAL_ACCESS_TOKEN }}
           ballerinaBotWorkflow: $ {{ secrets.BALLERINA_BOT_WORKFLOW }}
         run: |
-          ./gradlew build -Pversion=${VERSION}
+          ./gradlew build -Pversion=${VERSION} -x centralTests -x test
```
Comment (keizer619) [inline .github/workflows/publish-release.yml:76]:
```suggestion
          ./gradlew build -Pversion=${VERSION}
```

I hope these were skipped for testing work

Comment (Miranlfk) [reply .github/workflows/publish-release.yml]: Yes they were skipped for the purpose of testing I shall include them.

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -111,14 +111,35 @@ jobs:
...
+        run: |
+          npm install -g github-cli
+          gh --version
+      - name: Get Markdown file
+        id: file-url
+        env:
+          GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
+        run: |
+          gh repo view ballerina-platform/ballerina-dev-website --json url --jq '.clone_url'
+          gh api repos/ballerina-platform/ballerina-dev-website/contents/downloads/verification-notes/verification_notes_template.md -H 'Accept: application/vnd.github.v3.raw' > release_notes.md
```
Comment (keizer619) [inline .github/workflows/publish-release.yml:128]: @Miranlfk Shall we rename file name as `release-artfiacts-verification` as same name will be used to generate html web page name

Comment (Miranlfk) [reply .github/workflows/publish-release.yml]: The following change has been made

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -105,6 +109,38 @@ jobs:
...
+        id: file-url
+        env:
+          GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
+        run: |
+          gh repo view ballerina-platform/ballerina-dev-website --json url --jq '.clone_url'
+          gh api repos/ballerina-platform/ballerina-dev-website/contents/downloads/verification-notes/release-artfiacts-verification.md -H 'Accept: application/vnd.github.v3.raw' > release_notes.md
+      - name: Update Markdown file
+        run: |
+          npm install
+          node docs/update-version-notes/update_version.js
```
Comment (keizer619) [inline .github/workflows/publish-release.yml:132]: Lets replace with a sed command

File: .github/workflows/publish-release.yml [hunk 4]
```diff
@@ -69,7 +73,7 @@ jobs:
           devCentralToken: ${{ secrets.BALLERINA_DEV_CENTRAL_ACCESS_TOKEN }}
           ballerinaBotWorkflow: $ {{ secrets.BALLERINA_BOT_WORKFLOW }}
         run: |
-          ./gradlew build -Pversion=${VERSION}
+          ./gradlew build -Pversion=${VERSION}
```
Comment (keizer619) [inline .github/workflows/publish-release.yml:76]:
```suggestion
          ./gradlew build -Pversion=${VERSION}
```

File: .github/workflows/publish-release.yml (hunk 3 above)
Comment (Miranlfk) [reply .github/workflows/publish-release.yml]: This has been implemented

**Review by keizer619 — APPROVED** [review]

**Pull request merged**